presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -0,0 +1,156 @@
+import io.prestosql.spi.connector.ConnectorSplitSource;
+import io.prestosql.spi.connector.ConnectorTableHandle;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+import io.prestosql.spi.connector.FixedSplitSource;
+import io.prestosql.spi.connector.TableNotFoundException;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.message.BasicNameValuePair;
+
+import javax.inject.Inject;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static java.util.Objects.requireNonNull;
+
+public class PrometheusSplitManager
+        implements ConnectorSplitManager
+{
+    static final long DAYS_IN_A_YEAR = 365;
+    static final long DAYS_IN_A_WEEK = 7;
+    private final PrometheusClient prometheusClient;
+
+    @Inject
+    public PrometheusSplitManager(PrometheusClient prometheusClient)
+    {
+        this.prometheusClient = requireNonNull(prometheusClient, "client is null");
+        requireNonNull(prometheusClient.config.getQueryChunkSizeDuration(), "query-chunk-size-duration is null");
+        requireNonNull(prometheusClient.config.getMaxQueryRangeDuration(), "max-query-range-duration is null");
+    }
+
+    @Override
+    public ConnectorSplitSource getSplits(
+            ConnectorTransactionHandle transaction,
+            ConnectorSession session,
+            ConnectorTableHandle connectorTableHandle,
+            SplitSchedulingStrategy splitSchedulingStrategy)
+    {
+        PrometheusTableHandle tableHandle = (PrometheusTableHandle) connectorTableHandle;
+        PrometheusTable table = prometheusClient.getTable(tableHandle.getSchemaName(), tableHandle.getTableName());
+
+        // this can happen if table is removed during a query
+        if (table == null) {
+            throw new TableNotFoundException(tableHandle.toSchemaTableName());
+        }
+        List<ConnectorSplit> splits = new ArrayList<>();
+        generateTimesForSplits(Instant.now().toEpochMilli()).forEach(time -> {
+            try {
+                splits.add(new PrometheusSplit(buildQuery(prometheusClient.config.getPrometheusURI(), time, table.getName())));
+            }
+            catch (URISyntaxException e) {
+                throw new UnsupportedOperationException(e.getMessage());
+            }
+        });
+        return new FixedSplitSource(splits);
+    }
+
+    // URIBuilder handles URI encode
+    private URI buildQuery(URI baseURI, String time, String tableName)
+            throws URISyntaxException
+    {
+        List<NameValuePair> nameValuePairs = new ArrayList<>(2);
+        nameValuePairs.add(new BasicNameValuePair("query",
+                tableName + "[" + prometheusClient.config.getQueryChunkSizeDuration() + "]"));
```

Comment:
I feel like getQueryChunkSizeDuration should be an input parameter to this method since we're passing in the time. It makes it independent of the reliance on the prometheusClient as well, making it easier to test too.

Reply:
Done in 0901b40